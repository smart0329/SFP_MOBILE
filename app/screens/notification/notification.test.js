// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import Notification from './notification.js';

jest.mock('react-native-navigation', () => ({
    Navigation: {
        events: jest.fn(() => ({
            registerComponentDidDisappearListener: jest.fn(),
        })),
    },
}));

describe('Notification', () => {
    const baseProps = {
        actions: {
            loadFromPushNotification: jest.fn(),
        },
        componentId: 'component-id',
        deviceWidth: 100,
        notification: {},
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <Notification {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should call loadFromPushNotification on notification tap for non-local notification', () => {
        const props = {
            ...baseProps,
            notification: {
                localNotification: true,
            },
        };
        const wrapper = shallow(
            <Notification {...props}/>,
        );
        const instance = wrapper.instance();

        instance.notificationTapped();
        expect(baseProps.actions.loadFromPushNotification).not.toHaveBeenCalled();

        const notification = {
            localNotification: false,
        };
        wrapper.setProps({notification});
        instance.notificationTapped();
        expect(baseProps.actions.loadFromPushNotification).toHaveBeenCalledWith(notification);
    });
});
