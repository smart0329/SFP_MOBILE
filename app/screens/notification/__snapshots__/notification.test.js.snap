// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification should match snapshot 1`] = `null`;
